File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int i = digits.length - 1;
        while (i >= 0) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            //if digit is 9 then we make it 0 and loop continues till an digit less than 9 is found
            digits[i] = 0;
            i--;
        }
        int[] result = new int[digits.length+1];
        result[0] = 1;
        return result;
    }
}

File: 0066-plus-one/0066-plus-one.py
class Solution:
    def plusOne(self, digits: list[int]) -> list[int]:
        num=0
        for digit in digits:
            num= num*10 +digit
        num=num+1
        print(num)
        j=len(digits)+1
        print(j)
        answer=[0]*j
        print(answer)

        while(num>0):
            digit=num%10
            print("digit :",digit)
            answer[j-1]=digit
            j=j-1
            num=num//10
            print("num :",num)

        

        if answer[0]==0:
            answer.pop(0)
        print(answer)
        return answer